tarpon.gemspec

```diff
@@ -25,10 +25,12 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^bin/}) { |f| File.basename(f) }
   spec.require_paths = ['lib']
 
-  spec.required_ruby_version = '>= 2.7'
+  spec.required_ruby_version = '>= 2.6'
 
   spec.add_dependency 'http', '~> 4.4'
 
+  # last release before version 7 which requires Ruby 2.7. factory_bot needs activesupport and will go for v7 if no other is provided.
+  spec.add_development_dependency 'activesupport', '6.1.7'
```

Comment:
I added this because it was trying to install activesupport 7 (as a dependency of factorybot). 

I think it would resolve the right version if it's not a dependency of another library. 

This was the issue I was having specifically. 

However, I made this line dependent on having Ruby < 2.7 so it won't affect any of your previous users.

Reply:
Still can't get my head around why it is resolving to activesupport 7.
Are you using bundler? If so what's the content of your `Gemfile.lock`? What's the output of `ruby -v` inside your project directory?

Feel free to provide me a scenario where I can reproduce the issue.

Reply:
I believe it's resolving to activesupport 7 because factorybot depends on it and so will ask for that version unless there is another version installed. That's why I installed the "other" version: 

https://github.com/thoughtbot/factory_bot/tree/main/gemfiles

I don't see 6.2 here (like you have in your gemspec), but you can see factorybot 7 clearly specifies activesupport 7.

How would you like me to proceed?

Reply:
Again...I believe this is all for testing purposes. Should not affect your gem in production: 

https://github.com/thoughtbot/factory_bot/blob/main/gemfiles/7.0.gemfile

Reply:
> Again...I believe this is all for testing purposes. Should not affect your gem in production:
> 
> https://github.com/thoughtbot/factory_bot/blob/main/gemfiles/7.0.gemfile

I'm pretty sure it won't, I just want to understand the scenario as I still believe this check is not necessary.

ActiveSupport has a required ruby version of 2.7 or above for its version 7: https://rubygems.org/gems/activesupport/versions/7.0.0, so this version shouldn't be installed if you're running ruby 2.6.

What's the output of your `ruby -v` command inside the project directory?

Reply:
ruby 2.6.0p0 (2018-12-25 revision 66547) [x86_64-darwin18]

That's the output.

Reply:
Then `activesupport 7` shouldn't be installed as it explicitly requires ruby >= 2.7: https://github.com/rails/rails/blob/7-0-stable/activesupport/activesupport.gemspec#L12.

Could you add any more context (Gemfile, Gemfile.lock contents), or anything that could help me to understand your scenario.

I understand that the changes do not have impact for the others already using the gem, but I don't want to miss the opportunity to learn something.

Reply:
@Aryk following up here, did you have a chance to look at it?

Reply:
Hi @igorbelo I haven't looked at this again tbh, just juggling 500 things with Tribefy and it's just not priority. I'm still running this off the branch and it's working. I'll likely be upgrading my Ruby in a year so I'm just bunting on it. :)

You don't have to merge if you don't want, as this library is not very large and not changing very often. I can just stay on my branch for the meantime. :)

Reply:
Alright, feel free to bring this back if you need.